src/schema/v2/__tests__/marketingCollections.test.ts

```diff
@@ -172,4 +172,133 @@ describe("MarketingCollections", () => {
+        marketingCollections(sort: CURATED, size: 2, category: "Gallery") {
+          slug
+        }
+      }
+    `
+
+    const payload = [
+      {
+        slug: "percys-z-collection-1",
+      },
+      {
+        slug: "fiby-z-collection-2",
+      },
+    ]
+
+    const context = {
+      authenticatedLoaders: {},
+    } as any
+
+    context.marketingCollectionsLoader = jest
+      .fn()
+      .mockResolvedValue({ body: marketingCollectionsData })
+
+    const data = await runQuery(query, context)
+
+    expect(
+      context.marketingCollectionsLoader.mock.calls[0][0]
+    ).not.toContainKeys(["category", "sort"])
+
+    expect(context.marketingCollectionsLoader).toHaveBeenCalledWith({
+      size: 2,
+      slugs: [
+        "new-from-tastemaking-galleries",
+        "new-from-nonprofits-acaf27cc-2d39-4ed3-93dd-d7099e183691",
+        "new-from-small-galleries",
+        "new-from-leading-galleries",
+        "new-to-artsy",
+      ],
+    })
+
+    expect(data).toEqual({
+      marketingCollections: payload,
+    })
+  })
+
+  it("passes the params to the marketing collection loader", async () => {
+    const query = gql`
+      {
+        marketingCollections(
+          size: 2
+          category: "trending"
+          slugs: ["percys-z-collection-1", "fiby-z-collection-2"]
+        ) {
+          slug
+        }
+      }
+    `
+
+    const payload = [
+      {
+        slug: "percys-z-collection-1",
+      },
+      {
+        slug: "fiby-z-collection-2",
+      },
+    ]
+
+    const context = {
+      authenticatedLoaders: {},
+    } as any
+
+    context.marketingCollectionsLoader = jest
+      .fn()
+      .mockResolvedValue({ body: marketingCollectionsData })
+
+    const data = await runQuery(query, context)
+
+    expect(context.marketingCollectionsLoader).toHaveBeenCalledWith({
+      size: 2,
+      category: "trending",
+      slugs: ["percys-z-collection-1", "fiby-z-collection-2"],
+    })
+
+    expect(data).toEqual({
+      marketingCollections: payload,
+    })
+  })
+
+  it("throws an error when used with a non-existent category", async () => {
+    const query = gql`
+      {
+        marketingCollections(sort: CURATED, size: 2, category: "newest") {
+          slug
+        }
+      }
+    `
+    const context = {} as any
+    await expect(runQuery(query, context)).rejects.toThrow(
+      "No curated sort available for category: newest"
+    )
+  })
+
+  it("passes when used with a curated sort and without category", async () => {
+    const query = gql`
+      {
+        marketingCollections(sort: CURATED, size: 1) {
+          slug
+        }
+      }
+    `
+
+    const context = {
+      authenticatedLoaders: {},
+    } as any
+
+    context.marketingCollectionsLoader = jest
+      .fn()
+      .mockResolvedValue({ body: marketingCollectionsData })
+
+    await runQuery(query, context)
+
+    expect(
+      context.marketingCollectionsLoader.mock.calls[0][0]
+    ).not.toContainKeys(["sort", "category"])
```

Comment:
**suggestion**: I would expect this response to fail with an error but the test dosstring suggests that it would pass.

Generally, I feel like there could be three tests introduced
  - it("returns an ordered list of collection by category when CURATED sort is requested")
    - happy path
  - it("returns an error when CURATED sort is requested and category is missing")
  - it("returns an error when CURATED sort is requested and category is unknown")

Reply:
Yeah, that was the idea, but the test case naming was a bit misleading.